packages/db/src/browser.ts

```diff
@@ -1,13 +1,84 @@
-import { DataSource } from 'typeorm'
+import { BlockEntry, Database, KeyValueEntry } from '@acala-network/chopsticks-core'
+import { DBSchema, IDBPDatabase, openDB } from 'idb'
 
-import { BaseSqlDatabase } from './base-sql'
-import { openDb } from './db/browser'
-
-export class SqljsDatabase extends BaseSqlDatabase {
-  datasource: Promise<DataSource>
+interface Schema extends DBSchema {
+  keyValue: {
+    key: string
+    value: KeyValueEntry
+  }
+  block: {
+    key: string
+    value: BlockEntry
+    indexes: { byNumber: number }
+  }
+}
+export class IdbDatabase implements Database {
+  datasource: Promise<IDBPDatabase<Schema>>
 
   constructor(location: string) {
-    super()
-    this.datasource = openDb(location)
+    this.datasource = openDB<Schema>(location, 1, {
+      upgrade(db) {
+        db.createObjectStore('keyValue', { keyPath: 'key' })
+        const blockStore = db.createObjectStore('block', { keyPath: 'hash' })
+        blockStore.createIndex('byNumber', 'number')
+      },
+    })
+  }
+
+  async close(): Promise<void> {
+    const db = await this.datasource
+    db.close()
+  }
+
+  async saveBlock(block: BlockEntry): Promise<void> {
+    const db = await this.datasource
+    const tx = db.transaction(['block'], 'readwrite')
+    const store = tx.objectStore('block')
+    store.delete(block.hash)
+    store.put(block)
+    await tx.done
+  }
+
+  async queryBlock(hash: `0x${string}`): Promise<BlockEntry | null> {
+    const db = await this.datasource
+    const block = await db.get('block', hash)
+    return block ?? null
+  }
+
+  async queryBlockByNumber(number: number): Promise<BlockEntry | null> {
+    const db = await this.datasource
+    const block = await db.getFromIndex('block', 'byNumber', number)
+    return block ?? null
+  }
+
+  async queryHighestBlock(): Promise<BlockEntry | null> {
+    const db = await this.datasource
+    const block = await db.getAllFromIndex('block', 'byNumber')
```

Comment:
can't we just get one?

Reply:
`getFromIndex` needs to have the third argument